src/logic/fetch-elements/fetch-users-permissions.ts

```diff
@@ -0,0 +1,293 @@
+import { AppComponents, WalletLandPermission, PermissionType } from '../../types'
+
+type ParcelKey = `${string},${string}` // "x,y"
+
+/**
+ * Internal type for accumulating permissions from multiple sources
+ */
+type ParcelPermissionAccumulator = {
+  x: string
+  y: string
+  permissions: Set<PermissionType>
+  estate?: { id: string; size: number }
+}
+
+/**
+ * Query result types from LAND-permissions-graph subgraph
+ */
+type SubgraphParcel = {
+  x: string
+  y: string
+  tokenId: string
+  id: string
+}
+
+type SubgraphUser = {
+  parcels: SubgraphParcel[]
+  estates: Array<{
+    id: string
+    size: number
+    parcels: SubgraphParcel[]
+  }>
+}
+
+type SubgraphParcelWithOwner = SubgraphParcel & {
+  owner: { id: string }
+  estate?: { id: string; size: number }
+}
+
+type SubgraphEstate = {
+  id: string
+  size: number
+  owner: { id: string }
+  parcels: SubgraphParcel[]
+}
+
+type SubgraphAuthorization = {
+  owner: {
+    parcels: Array<
+      SubgraphParcel & {
+        estate?: { id: string; size: number }
+      }
+    >
+    estates: Array<{
+      id: string
+      size: number
+      parcels: SubgraphParcel[]
+    }>
+  }
+}
+
+type QueryResult = {
+  user: SubgraphUser | null
+  updateOperatorParcels: SubgraphParcelWithOwner[]
+  updateOperatorEstates: SubgraphEstate[]
+  updateManagerAuthorizations: SubgraphAuthorization[]
+}
+
+/**
+ * Comprehensive GraphQL query that retrieves all authorization types for a user
+ */
+const QUERY_USER_PERMISSIONS = `
+query GetAllUserPermissions($address: String!) {
+  # 1. Direct parcel and estate ownership
+  user: wallet(id: $address) {
+    parcels {
+      x
+      y
+      tokenId
+      id
+    }
+    estates {
+      id
+      size
+      parcels {
+        x
+        y
+        tokenId
+        id
+      }
+    }
+  }
+
+  # 2. Parcel-level UpdateOperator permissions
+  updateOperatorParcels: parcels(
+    where: { updateOperator: $address }
+    first: 1000
+  ) {
+    x
+    y
+    tokenId
+    id
+    owner { id }
+    estate { id, size }
+  }
+
+  # 3. Estate-level UpdateOperator permissions
+  updateOperatorEstates: estates(
+    where: { updateOperator: $address }
```

Comment:
There's the `updateOperator` and the `operator` for estates, we should retrieve all estates that match any of these properties, would you mind changing the query to match that?